tron/mcp.py

```diff
@@ -115,8 +115,13 @@ def build_job_scheduler_factory(self, master_config):
             output_stream_dir,
             master_config.time_zone,
             action_runner,
+            self.eventbus_publish,
         )
 
+    def eventbus_publish(self, message):
```

Comment:
just to repeat what we spoke about offline:

I wonder if it might make for a cleaner interface if, rather than sending this callback down the stack, we expose a Queue object that the eventbus is on the other end of. It means you can cleanly seperate the consumer from the the publisher (so you can replace the eventbus object at will in the mcp), and I'd argue that it's less complex than dealing with callbacks as you've done here. 

The downside is that you'd have to create a NoopEventBus or something to consume off the queue and do nothing with the events to avoid the queue filling up.